docs/content/foundation/box-primitive.mdx

```diff
@@ -0,0 +1,75 @@
+---
+title: Box Primitive
+---
+
+import Link from 'gatsby-link';
+import { useStyles } from '@marigold/system';
+
+# Box Primitive
+
+## Box as Layout Component
+
+As the basis of all our components, we use the `<Box>`. It is a versatile layout component that
+can be modified with layout props as listed in the <Link to="../../components/box">Box component</Link> description.
+Additionally, use the `variant` prop to define custom styles in your themes. With the `as` prop, you
+determine the rendered HTML element. Instead of exposing the underlying tools that will create CSS, the `<Box>` component
+wraps them and exposes an API for applying styling to a component while respecting the rules and constraints of our design
+system.
+
+```tsx code
+import { Box } from '@marigold/components';
+import { useStyles } from '@marigold/system';
+```
+
+```tsx live
+<Box as="button" bg="orange" p="8px">
+  Button box with backgroundcolor and padding!
+</Box>
+```
+
+## Building a custom Component with the Box
+
+In case you want to build your own custom component, use the `Box` as the underlying layout component and add
+the styles accordingly. You can stack and nest multiple Box instances and build your complex component like this.
+By default, the `Box` component will render a `<div />`. This might be fine most of the times, but sometimes you would
+rather render a `<button>` or an `<input>` etc. To allow this, the `Box` takes the special prop called `as`, accepting
+all HTML tag names as input. Don't forget to define variants in your theme.
+
+```tsx code
+export type ButtonProps = {
+  variant?: string;
+} & ComponentPropsWithRef<'button'>;
+
+export const Button: React.FC<ButtonProps> = ({
+  variant = 'primary.large',
+  children,
+  ...props
+}) => {
+  return (
+    <Box as="button" {...props} variant={`button.${variant}`}>
+      <Box as="span" display="inline-flex" alignItems="center">
+        {children}
+      </Box>
+    </Box>
+  );
+};
+```
+
+## Custom CSS
+
+Staying inside the boundaries of the design system and its contraints should be the norm. But as limiting styling only
+to allowed values defined by the system can be very restrictive, you can use the `className` prop and generate custom
+styles with the <Link to="../../guides/usestyles">`useStyles` hook</Link>.
```

Comment:
Just a little thing. I would take the word hook out of the link. Looks a little bit komisch 😀